### ISSUE TITLE
Markdown (.md) files contain an extra newline when writing when a title node body is 'empty'

### ISSUE THREAD

**Issue by boltex** [leo-editor/leo-editor#3719]
**EKR**: As noted in PR #3722, this is a round-tripping problem.

**Original Report**

The problem happens when writing @auto nodes of markdown content. (reading seems fine)

The problem arises only if a title node **without body text** has a child of a lower title. 

For example, this valid markdown will be read properly but an extra newline will be inserted between **# 1st level title A** and **## 2nd level title B** when re-written to disk with **ctrl+shift+w** _(it will have 2 empty lines between instead of one)_

Original markdown: (note the absence of text in node "# 1st level title A, which will provoke the double newline insertion when written to file)
```
# 1st level title X

some text in body X

## 2nd level title Z

some text in body Z

# 1st level title A

## 2nd level title B 

some body content of the 2nd node 
```
I understand that when writing a newline is placed below the headline which generates the title (with the proper number of '#', but perhaps a check could be made to see if the body pane contained **specifically nothing, or a single newline**, if so, it should skip writing that extra newline.

Here's a visual example of the problem:

Leo reads [the original markdown](https://github.com/leo-editor/leo-editor/assets/847716/b52cb193-2fd1-4a94-998c-e4435425941b) correctly as shown  [here](https://github.com/leo-editor/leo-editor/assets/847716/cb58e294-2a05-418f-bdc9-60eeef42e242).

But after a write [the 'empty' section now has an extra newline](https://github.com/leo-editor/leo-editor/assets/847716/3b325ed2-0a72-4627-bd4f-8e54ed8081d0). _Note where the cursor row is highlighted_.

The whole read - write cycle would be perfect without this single small annoyance. 

Markdown has become the #1 standard for documentation, so I think this is important.

**Comment by boltex** [comment]
I've spotted why the test didn't work: I've annotated the PR.

While annotating the PR I thought that perhaps the problem could be solved upon reading/refreshing the @auto node instead of when writing it. As I've observed that an extra newline is present in the '1st level title A' node's body upon reading from file. 

So this issue could perhaps be 'fixed' at the reading (importer) of the @auto file instead of the writing. (writer) But I'm not sure about that, just a thought I had. :)
